src/components/periodic-table/PeriodicTable.tsx

```diff
@@ -1,177 +1,101 @@
+      <div className="periodic-table__cell periodic-table__cell--label">4</div>
+      <div className="periodic-table__cell periodic-table__cell--label">5</div>
+      <div className="periodic-table__cell periodic-table__cell--label">6</div>
+      <div className="periodic-table__cell periodic-table__cell--label">7</div>
+      <div className="periodic-table__cell periodic-table__cell--label">8</div>
+      <div className="periodic-table__cell periodic-table__cell--label">9</div>
+      <div className="periodic-table__cell periodic-table__cell--label">10</div>
+      <div className="periodic-table__cell periodic-table__cell--label">11</div>
+      <div className="periodic-table__cell periodic-table__cell--label">12</div>
+      <div className="periodic-table__cell periodic-table__cell--label">13</div>
+      <div className="periodic-table__cell periodic-table__cell--label">14</div>
+      <div className="periodic-table__cell periodic-table__cell--label">15</div>
+      <div className="periodic-table__cell periodic-table__cell--label">16</div>
+      <div className="periodic-table__cell periodic-table__cell--label">17</div>
+      <div className="periodic-table__cell periodic-table__cell--label">18</div>
+    </div>,
+  ];
+
+  for (let i = 0; i < periodicTableData.length; i++) {
+    const row = periodicTableData[i];
+
+    rows.push(
+      <div className="periodic-table__row" key={`row-${i}`}>
+        {i <= 6 ? (
+          <div className="periodic-table__cell periodic-table__cell--label">
+            {i + 1}
+          </div>
+        ) : (
+          <div className="periodic-table__cell periodic-table__cell--label" />
+        )}
+
+        {row.map((element, index) => {
+          if (element <= 0) {
+            return (
+              <div
+                key={`row-${i}-cell-${index}`}
+                className="periodic-table__cell periodic-table__cell--empty"
+              />
+            );
+          }
+
+          return (
+            <div
+              className="periodic-table__cell"
+              key={`row-${i}-cell-${index}`}
+            >
+              {elementRenderer(element)}
+            </div>
+          );
+        })}
+      </div>
+    );
+  }
 
-  constructor(props: IPeriodicTableProps) {
-    super(props);
+  return rows;
+}
 
-    this.periodicTableData = data;
-  }
+function PeriodicTable({ elementRenderer }: IPeriodicTableProps) {
+  const [render, setRender] = React.useState(false);
 
-  public componentDidMount() {
-    this.requestAnimationFrame = window.requestAnimationFrame(
+  React.useEffect(() => {
+    let requestAnimationFrame: number = window.requestAnimationFrame(
       () =>
-        (this.requestAnimationFrame = window.requestAnimationFrame(() =>
-          this.setState({ render: true })
+        (requestAnimationFrame = window.requestAnimationFrame(() =>
+          setRender(true)
         ))
     );
+    return () => window.cancelAnimationFrame(requestAnimationFrame);
+  }, []);
```

Comment:
This is more a question for @HorusGoul, but why do we request an animation from in order to render this component?

Reply:
I didn't understand too. Maeby to show a little spinner before loading the page ❓